controllers/users.js

```diff
@@ -291,7 +291,19 @@ const updateSelf = async (req, res) => {
       await userQuery.setIncompleteUserDetails(userId);
     }
 
-    const user = await userQuery.addOrUpdate(req.body, userId);
+    let updatedUserData = req.body;
+    if (req.body.discordId) {
```

Comment:
If someone not in our discord sends a random discord id wouldn't this set their in_discord to true